make/git.mk

```diff
@@ -26,5 +26,13 @@ bin/scratch/git/upstream-tags.txt: | bin/scratch/git
 		sed -n '/v1.0.0/,$$p' | \
 		grep -v "v1.2.0-alpha.1" > $@
 
+# The file "release-version" gets updated whenever git describe --tags changes. This is
+# used by the bin/containers/*.tar.gz targets to make sure that the containers, which use
+# the output of "git describe --tags" as their tag, get rebuilt whenever you check out a
+# different commit. If we didn't do this, the Helm chart bin/cert-manager-*.tgz would refer
+# to an image tag that doesn't exist in bin/containers/*.tar.gz.
+bin/release-version: | bin
+	@test "$(RELEASE_VERSION)" == "$$(cat "$@" 2>/dev/null)" || echo $(RELEASE_VERSION) > $@
```

Comment:
You are correct, this is a `.PHONY` target, not a timestamp-based target. You are also right in that the test `$(RELEASE_VERSION)" == "$$(cat "$@" 2>/dev/null)` will systematicaly be re-run on every invocation of `bin/release-version`.

For example, when running:

```sh
make -f make/Makefile all-containers
```

the target will be only ever called once, even if it is a prerequisite for 20 different targets.

The reason I bother is that I was surprised multiple times with the edge case of committing after building the containers. I build the container, commit, run  `helm upgrade --install --set image.tag=$(git describe --tags)` and I unexpectedly don't see the changes I was expecting to see.

Another way to work around this could be to do what we did with bazel: tagging with a timestamp.

Reply:
> You are correct, this is a .PHONY target, not a timestamp-based target. 

But this _isn't_ a PHONY target, is it? It's not marked `.PHONY` anywhere that I can see?

And the issue is that if we _do_ mark it `.PHONY` it will force-rebuild everything which has it as a dependency, on every run.

Reply:
Haha I think you pushed right as I sent that comment, disregard!

Reply:
You were correct, this file wasn't updated properly, I have added `.PHONY` (and read the manual).

> And the issue is that if we do mark it .PHONY it will force-rebuild everything which has it as a dependency, on every run.

I agree. If the target was:

```make
bin/release-version: | bin
     echo $(RELEASE_VERSION) > @$
```

then all the targets having `bin/release-version` as their prerequisite would constantly get re-run.

That's why I have added the test; it should only ever change (timestamp-wise) when `RELEASE_VERSION` changes.

Reply:
Ah, I think `.PHONY` isn't a great idea, since it disables the "newer timestamp, must rebuild"mechanism on the `bin/release-version`.

That means that any target having `bin/release-version` as prerequisite gets rebuild systematically.

As the manual suggest, I'll use the `FORCE` approach. This time, the "newer timestamp to rebuild" works.

Reply:
FORCE is _much_ better yeah, thank you. Stops us forcing a rebuild even when nothing changes. :+1: 

I'll re-review now!